src/libsync/propagateuploadng.cpp

```diff
@@ -262,7 +262,42 @@ void PropagateUploadFileNG::startNextChunk()
 
     quint64 fileSize = _item->_size;
     Q_ASSERT(fileSize >= _sent);
-    quint64 currentChunkSize = qMin(chunkSize(), fileSize - _sent);
+
+    quint64 currentChunkSize = chunkSize();
+
+    // this will check if getRequestMaxDurationDC is set to 0 or not
+    double requestMaxDurationDC = (double) getRequestMaxDurationDC();
+    if (requestMaxDurationDC != 0) {
+        // this if first chunked file request, so it can start with default size of chunkSize()
+        // if _lastChunkSize != 0 it means that we already have send one request
+        if(_lastChunkSize != 0){
+            //TODO: this is done step by step for debugging purposes
```

Comment:
Please leave it "by step". The compiler will anyway optimize. It is very very very important that we also understand this code in some months.